truss/base/truss_config.py

```diff
@@ -273,24 +273,28 @@ def from_dict(d):
         use_gpu = d.get("use_gpu", DEFAULT_USE_GPU)
         if accelerator.accelerator is not None:
             use_gpu = True
-        # TODO[rcano]: add validation for node count
-        node_count = d.get("node_count", DEFAULT_NODE_COUNT)
-
-        return Resources(
-            cpu=cpu,
-            memory=memory,
-            use_gpu=use_gpu,
-            accelerator=accelerator,
-            node_count=node_count,
-        )
+
+        r = Resources(cpu=cpu, memory=memory, use_gpu=use_gpu, accelerator=accelerator)
+
+        # only add node_count if not None. This helps keep
+        # config generated by truss init concise.
+        node_count = d.get("node_count")
+        if node_count is not None:
+            validate_node_count(node_count)
+            r.node_count = node_count
```

Comment:
I don't think we have to do this logic in `from_dict` right? I think we can just do:

```
        node_count = d.get("node_count")
        if node_count is not None:
            validate_node_count(node_count)
        return Resources(cpu=cpu, memory=memory, use_gpu=use_gpu, accelerator=accelerator, node_count=node_count)
```

If `node_count` is `None`, it'll be None anyway